java/google/registry/rde/imports/RdeHostLinkAction.java

```diff
@@ -0,0 +1,162 @@
+// Copyright 2017 The Nomulus Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package google.registry.rde.imports;
+
+import static com.google.common.base.Preconditions.checkState;
+import static google.registry.mapreduce.MapreduceRunner.PARAM_MAP_SHARDS;
+import static google.registry.model.EppResourceUtils.loadByForeignKey;
+import static google.registry.model.ofy.ObjectifyService.ofy;
+import static google.registry.model.registry.Registries.findTldForName;
+import static google.registry.util.PipelineUtils.createJobPath;
+
+import com.google.appengine.tools.mapreduce.Mapper;
+import com.google.common.base.Joiner;
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Iterables;
+import com.google.common.net.InternetDomainName;
+
+import com.googlecode.objectify.Key;
+import com.googlecode.objectify.VoidWork;
+import google.registry.config.RegistryConfig.Config;
+import google.registry.mapreduce.MapreduceRunner;
+import google.registry.model.domain.DomainResource;
+import google.registry.model.host.HostResource;
+import google.registry.request.Action;
+import google.registry.request.Parameter;
+import google.registry.request.Response;
+import google.registry.util.FormattingLogger;
+import google.registry.xjc.JaxbFragment;
+import google.registry.xjc.rdehost.XjcRdeHost;
+import google.registry.xjc.rdehost.XjcRdeHostElement;
+import org.joda.time.DateTime;
+
+import javax.annotation.Nullable;
+import javax.inject.Inject;
+
+/**
+ * A mapreduce that links hosts from an escrow file to their superordinate domains.
```

Comment:
Can you expand on this a little bit more?  If import is a 4-step process of mapreduces, which step is this?  I'd also say something along the lines of this is run after domains and hosts have been imported separately, it runs over all hosts, checks to see if each one is subordinate (i.e. the host name with the first label stripped off is a domain name that exists in a TLD run by this registry), and if so, modifies the superordinateDomain field on the host to point to the domain.

Also some justification as to why this step is needed separately.  Something like: Hosts have to be imported before domains because domains point to hosts, so when hosts are initially imported the domains don't yet exist.  So we need to go back after the domains are imported and then edit the hosts again to point to their superordinate domains.

I assume that's correct?